src/Doctrine/Common/State/Options.php

```diff
@@ -22,6 +22,7 @@ class Options implements OptionsInterface
      */
     public function __construct(
         protected mixed $handleLinks = null,
+        protected mixed $transformModel = null,
```

Comment:
I like the feature, I think it'd be a great fit with https://github.com/api-platform/core/pull/6801 best would be that when the mapper is installed, we add a default transformer that uses the object mapper! 

I'm just not a fan of the naming but this one is hard haha. A few ideas:
 - transformEntityToResource 
 - entityToResourceMapper
 - mapper
 - resourceTransformer
 
I think that it's missing a way to transform a resource to an entity when persisting data (POST/PATCH) WDYT? 

Last but not least, you should work on an interface that'd be used to autoconfigure the "transformEntityLocator", it would surely help with naming things.